packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -75,57 +73,124 @@ const invalidTransferRoots: Record<string, boolean> = {
   '0xf902d5143ceee334fce5d56483024e0f4c476a1b5065d9d39d6c1deb6513b7bb': true
 }
 
-class TransferRootsDb extends TimestampedKeysDb<TransferRoot> {
+class TransferRootsDb extends BaseDb {
   subDbIncompletes: any
+  subDbTimestamps: any
+  subDbRootHashes: any
 
   constructor (prefix: string, _namespace?: string) {
     super(prefix, _namespace)
+
+    this.subDbTimestamps = new BaseDb(`${prefix}:timestampedKeys`, _namespace)
     this.subDbIncompletes = new BaseDb(`${prefix}:incompleteItems`, _namespace)
+    this.subDbRootHashes = new BaseDb(`${prefix}:rootHashes`, _namespace)
+
+    this.migration()
+      .then(() => {
+        this.ready = true
+        this.logger.debug('db ready')
+      })
+      .catch(err => {
+        this.logger.error(err)
+      })
+  }
+
+  async migration () {
+    // re-index timestamped db keys by transfer root id
+    let items = await this.subDbTimestamps.getKeyValues()
+    let promises: Array<Promise<any>> = []
+    for (const { key, value } of items) {
+      promises.push(new Promise(async (resolve) => {
+        if (!value?.transferRootHash) {
+          resolve(null)
+          return
+        }
+        const item = await this.getById(value.transferRootHash)
```

Comment:
Does this (and some of the same types of calls on subsequent lines) need to `getById` from the specific DB? Such as `this.subDbTimetamps.getById()`?

Reply:
No necessarily; `this.getById` refers to the main db with all the entries and we need to query the entry by transferRootHash (current key) to find the transferRootId in order to use that as the new key.